server/lib/report_server/clue.ex

```diff
@@ -0,0 +1,207 @@
+      question_id = String.downcase(tile_title)
+      username = row["username"]
+      [user_id, portal_site] = String.split(username, "@")
+      portal_url = "https://#{portal_site}"
+      learner = Enum.find(learners, fn learner -> Integer.to_string(learner.user_id) == user_id end)
+
+      updated_questions = Map.put(row_acc.structure.questions, question_id, %{
+        :type => "clue_text_tile",
+        :prompt => tile_title,
+        :required => false
+      })
+      updated_question_order = [question_id | row_acc.structure.question_order]
+
+      url = HistoryLink.format_link_to_work(%HistoryLink{
+        portal_url: portal_site,
+        offering_id: Integer.to_string(learner.offering_id),
+        class_id: Integer.to_string(learner.class_id),
+        document_key: row["document_key"],
+        document_uid: user_id,
+        maybe_document_history_id: row["document_history_id"]})
+
+      updated_answers = with text_field <- row["text_value"],
+            text_trimmed <- String.trim_leading(text_field, "\"") |> String.trim_trailing("\""),
+            {:ok, json} <- Jason.decode(text_trimmed),
+            plain_text <- extract_text(json),
+            {:ok, answer_json} <- Jason.encode(%{ "text" => plain_text, "url" => url }) do
+        answer_row = %{
+          question_id: question_id,
+          answer: answer_json,
+          platform_user_id: user_id,
+          resource_link_id: Integer.to_string(learner.offering_id),
+          remote_endpoint: learner.run_remote_endpoint,
+          id: row["id"], ## Using the ID of the event here, but it could be any arbitrary ID
+          ## The following are constant and could be added later
+          resource_url: url,
+          platform_id: portal_url,
+          source_key: "collaborative-learning.concord.org",
+          tool_id: "collaborative-learning.concord.org",
+          version: "1",
+          submitted: false,
+          run_key: nil,
+          context_id: nil,
+          class_info_url: nil,
+          type: nil,
+          question_type: nil,
+          tool_user_id: nil,
+          created: nil
+        }
+        user_answers = [ answer_row | Map.get(row_acc.answers, username, [])]
+        Map.put(row_acc.answers, username, user_answers)
+      else
+        _ -> row_acc.answers
+      end
+
+      %{
+        structure: %{
+          questions: updated_questions,
+          choices: %{},
+          question_order: updated_question_order
+        },
+        answers: updated_answers
+      }
+    end)
+
+    ## Loop over answers and write a parquet file for each username
+    write_attempts = Enum.map(result.answers, fn {username, answerlist} ->
+      resource_link_id = answerlist |> List.first() |> Map.get(:resource_link_id)
+      with {:ok, path} <- get_parquet_file_path(url, username, resource_link_id) do
+        answers_df = Explorer.DataFrame.new(answerlist)
+        Explorer.DataFrame.to_parquet(answers_df, path)
```

Comment:
The path here is an `s3://` path right?  Writing to the filesystem directly isn't going to work on production as the ECS docker containers can't write files locally.